### PR TITLE
Create decorator for exception handling for secrets/iac/sca commands

### PR DESCRIPTION
**Pull request by amascia-gg**
Remove try/except blocks in commands from Secrets / IaC / SCA in favour of a decorator

### CODE REVIEW

**Review by GG-HH — COMMENTED** [review]
Ok to me, I am a bit confused by the `config: Config = args[0].obj["config"]` which seems a bit ugly to me.

**Comment by GG-HH** [comment]
@agateau-gg WDYT ? Would you prefer if we remain with try/except blocks everywhere to be more explicit ?

File: ggshield/cmd/utils/common_decorators.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            return handle_exception(error, config.user_config.verbose)
+
+    return wrapper
+
+
+def display_beta_warning(func):
+    """
+    Displays warning about new verticals' commands being in beta.
+    """
+
```
Comment (agateau-gg) [inline ggshield/cmd/utils/common_decorators.py:27]: Should probably use `@wraps` here too, unless I am missing something.

**Review by agateau-gg — CHANGES_REQUESTED** [review]
I like the idea of reducing indentation level, and the fact that the decorate means a bit less boilerplate ✨.

Would be even better to wrap the call to `cli.main()` in `ggshield.__main__.main()` inside a try..except block instead of having to decorate all functions, but I am not sure this is possible.

"Request changes" because of a minor remark in the beta decorator.

**Comment by amascia-gg** [comment]
> I like the idea of reducing indentation level, and the fact that the decorate means a bit less boilerplate ✨.
> 
> Would be even better to wrap the call to `cli.main()` in `ggshield.__main__.main()` inside a try..except block instead of having to decorate all functions, but I am not sure this is possible.
> 
> "Request changes" because of a minor remark in the beta decorator.

The context does not exist at this stage just yet so it was not possible unfortunately.

**Review by agateau-gg — APPROVED** [review]
Looks good, thanks!

**Pull request merged**